pnpm-{audit, outdated}: add page for pnpm review dependencies (#12437)
tldr-pages/tldr: pages/common/pnpm-audit.md

# pnpm audit

> > Scans project dependencies.
> > Checks for known security issues with the installed packages.

Reviewer: ```suggestion
> Scan project dependencies.
> Check for known security issues with the installed packages.
```

> More information: <https://pnpm.io/cli/audit>.

- Identify vulnerabilities in the project:

`pnpm audit`

- Automatically fix vulnerabilities:

`pnpm audit fix`

- Generate a security report in JSON format:

> `pnpm audit --json > audit-report.json`

Reviewer 2: ```suggestion
`pnpm audit --json > {{path/to/audit-report.json}}`
```

- Audit only [D]ev dependencies:

`pnpm audit -D`

> - Audit only production dependencies:

Reviewer 2: ```suggestion
- Audit only [P]roduction dependencies:
```

> `pnpm audit -D`
> 
> - Audit only [P]roduction dependencies:
> 
> `pnpm audit -P`

Reviewer: ```suggestion
`pnpm audit --dev`

- Audit only [P]roduction dependencies:

`pnpm audit --prod`
```
Since we highlight the short option as mnemonics, maybe we could use the long option here.

> - Exclude optionalDependencies from the audit:

Reviewer 2: ```suggestion
- Exclude optional dependencies from the audit:
```

`pnpm audit --no-optional`

- Ignore registry errors during the audit process:

`pnpm audit --ignore-registry-errors`

- Filter advisories by severity (low, moderate, high, critical):

> `pnpm audit --audit-level <severity>`

Reviewer 2: ```suggestion
`pnpm audit --audit-level {{severity}}`
```
